### ISSUE TITLE
cffi-toolchain not compiling

### ISSUE THREAD

**Issue by duncan-bayne** [quicklisp/quicklisp-projects#1418]
When trying to load `cffi-toolchain` with:

```
CL-USER> (ql:quickload "cffi-toolchain")
```

... I get:

```
COMPILE-FILE-ERROR while
compiling #<CL-SOURCE-FILE "cffi-toolchain" "toolchain" "bundle">
   [Condition of type UIOP/LISP-BUILD:COMPILE-FILE-ERROR]
...
Backtrace:
  0: (UIOP/LISP-BUILD:CHECK-LISP-COMPILE-RESULTS NIL T T "~/asdf-action::format-action/" ((#<ASDF/LISP-ACTION:COMPILE-OP > . #<ASDF/LISP-ACTION:CL-SOURCE-FILE "cffi-toolchain" "toolchain" "bundle">)))
  1: ((SB-PCL::EMF ASDF/ACTION:PERFORM) #<unavailable argument> #<unavailable argument> #<ASDF/LISP-ACTION:COMPILE-OP > #<ASDF/LISP-ACTION:CL-SOURCE-FILE "cffi-toolchain" "toolchain" "bundle">)
  2: ((:METHOD ASDF/ACTION:PERFORM-WITH-RESTARTS :AROUND (T T)) #<ASDF/LISP-ACTION:COMPILE-OP > #<ASDF/LISP-ACTION:CL-SOURCE-FILE "cffi-toolchain" "toolchain" "bundle">) [fast-method]
  3: ((:METHOD ASDF/PLAN:PERFORM-PLAN (LIST)) ((#1=#<ASDF/LISP-ACTION:PREPARE-OP > . #2=#<ASDF/SYSTEM:SYSTEM "asdf">) (#3=#<ASDF/LISP-ACTION:COMPILE-OP > . #2#) (#4=#<ASDF/LISP-ACTION:LOAD-OP > . #2#) (#1#..
  4: ((FLET SB-C::WITH-IT :IN SB-C::%WITH-COMPILATION-UNIT))
  5: ((:METHOD ASDF/PLAN:PERFORM-PLAN :AROUND (T)) ((#1=#<ASDF/LISP-ACTION:PREPARE-OP > . #2=#<ASDF/SYSTEM:SYSTEM "asdf">) (#3=#<ASDF/LISP-ACTION:COMPILE-OP > . #2#) (#4=#<ASDF/LISP-ACTION:LOAD-OP > . #2#)..
  6: ((FLET SB-C::WITH-IT :IN SB-C::%WITH-COMPILATION-UNIT))
  7: ((:METHOD ASDF/PLAN:PERFORM-PLAN :AROUND (T)) #<ASDF/PLAN:SEQUENTIAL-PLAN {10079FF993}> :VERBOSE NIL) [fast-method]
  8: ((:METHOD ASDF/OPERATE:OPERATE (ASDF/OPERATION:OPERATION ASDF/COMPONENT:COMPONENT)) #<ASDF/LISP-ACTION:LOAD-OP :VERBOSE NIL> #<ASDF/SYSTEM:SYSTEM "cffi-toolchain"> :VERBOSE NIL) [fast-method]
  9: ((SB-PCL::EMF ASDF/OPERATE:OPERATE) #<unused argument> #<unused argument> #<ASDF/LISP-ACTION:LOAD-OP :VERBOSE NIL> #<ASDF/SYSTEM:SYSTEM "cffi-toolchain"> :VERBOSE NIL)
 10: ((LAMBDA NIL :IN ASDF/OPERATE:OPERATE))
 11: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (T T)) #<ASDF/LISP-ACTION:LOAD-OP :VERBOSE NIL> #<ASDF/SYSTEM:SYSTEM "cffi-toolchain"> :VERBOSE NIL) [fast-method]
 12: ((SB-PCL::EMF ASDF/OPERATE:OPERATE) #<unused argument> #<unused argument> ASDF/LISP-ACTION:LOAD-OP "cffi-toolchain" :VERBOSE NIL)
 13: ((LAMBDA NIL :IN ASDF/OPERATE:OPERATE))
 14: (ASDF/CACHE:CALL-WITH-ASDF-CACHE #<CLOSURE (LAMBDA NIL :IN ASDF/OPERATE:OPERATE) {10079DA90B}> :OVERRIDE NIL :KEY NIL)
 15: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (T T)) ASDF/LISP-ACTION:LOAD-OP "cffi-toolchain" :VERBOSE NIL) [fast-method]
 16: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (T T)) ASDF/LISP-ACTION:LOAD-OP "cffi-toolchain" :VERBOSE NIL) [fast-method]
 17: (ASDF/OPERATE:LOAD-SYSTEM "cffi-toolchain" :VERBOSE NIL)
 18: (QUICKLISP-CLIENT::CALL-WITH-MACROEXPAND-PROGRESS #<CLOSURE (LAMBDA NIL :IN QUICKLISP-CLIENT::APPLY-LOAD-STRATEGY) {100795C3FB}>)
 19: (QUICKLISP-CLIENT::AUTOLOAD-SYSTEM-AND-DEPENDENCIES "cffi-toolchain" :PROMPT NIL)
```
Relevant versions:
```
$ ros --version
roswell 17.11.10.85(b0be720)

$ ros config
slime.version=2017.11.30
ccl-bin.version=1.11
setup.time=3722552654
sbcl-bin.version=1.2.7
default.lisp=sbcl-bin
```

**Comment by quicklisp** [comment]
That is a very old SBCL - does the problem remain with a current SBCL?

**Comment by guicho271828** [comment]
It loads fine on my environment

```
CL-USER> (ql:quickload :cffi-toolchain)
To load "cffi-toolchain":
  Load 1 ASDF system:
    cffi-toolchain
; Loading "cffi-toolchain"
.
(:CFFI-TOOLCHAIN)
CL-USER> (lisp-implementation-version)
"1.4.1"
CL-USER> (lisp-implementation-type)
"SBCL"
```

**Comment by duncan-bayne** [comment]
@quicklisp It's the default binary version of SBCL installed by Roswell; to be honest I'd just assumed it was somewhat current, but clearly not.  I'll try again building SBCL 1.3.21 from source (again via Roswell) and try again.  I'll update (close) this issue if it succeeds.

**Comment by duncan-bayne** [comment]
Fixed on 1.3.21.

**Comment by duncan-bayne** [comment]
Merry Christmas :)
